apps/vault/contracts/Vault.sol

```diff
@@ -1,118 +1,103 @@
 pragma solidity 0.4.18;
 
-import "./VaultBase.sol"; // split made to avoid circular import
+import "@aragon/os/contracts/apps/AragonApp.sol";
 
-import "./connectors/ERC20Connector.sol";
-import "./connectors/ETHConnector.sol";
+import "@aragon/os/contracts/lib/zeppelin/token/ERC20.sol";
 
-import "@aragon/os/contracts/lib/misc/Migrations.sol";
 
+contract Vault is AragonApp {
+    bytes32 constant public TRANSFER_ROLE = keccak256("TRANSFER_ROLE");
 
-contract Vault is VaultBase {
-    struct TokenStandard {
-        uint32 erc;
-        uint32 interfaceDetectionERC;
-        bytes4 interfaceID;
-        address connector;
-    }
-
-    TokenStandard[] public standards;
-    mapping (address => address) public connectors;
-    mapping (uint32 => bool) public supportedInterfaceDetectionERCs;
-
-    // this is required to avoid having to redeploy on every instance.
-    // they can just be gotten from the base by templates
-    ERC20Connector public erc20ConnectorBase;
-    ETHConnector public ethConnectorBase;
-
-    event NewTokenStandard(uint32 indexed erc, uint32 indexed interfaceDetectionERC, bytes4 indexed interfaceID, address connector);
+    event Transfer(address indexed token, address indexed to, uint256 amount);
+    event Deposit(address indexed token, address indexed sender, uint256 amount);
 
-    function Vault() public {
-        erc20ConnectorBase = new ERC20Connector();
-        ethConnectorBase = new ETHConnector();
+    /**
+    * @dev When used behind an AppProxy, this fallback is never executed
+    *      as it will be intercepted by the Proxy (see aragonOS#281)
+    */
+    function () payable external {
+        require(msg.data.length == 0);
+        deposit(ETH, msg.sender, msg.value);
     }
 
-    function initialize(ERC20Connector erc20Connector, ETHConnector ethConnector) onlyInit public {
-        initialized();
-
-        supportedInterfaceDetectionERCs[NO_DETECTION] = true;
-        supportedInterfaceDetectionERCs[ERC165] = true;
+    /**
+    * @notice Deposit `value` `token` to the vault
+    * @param token Address of the token being transferred
+    * @param from Entity that currently owns the tokens
+    * @param value Amount of tokens being transferred
+    */
+    function deposit(address token, address from, uint256 value) payable public {
+        require(value > 0);
+
+        if (token == ETH) {
+            // Deposit is implicit in this case
+            require(msg.sender == from && msg.value == value);
```

Comment:
If we remove the `msg.sender == from` check, we can pass along the actual sender when other apps deposit to the vault (see Finance, above).

Reply:
I'm having some doubts about this. Given that people may create an allowance greater than the funds they want to deposit to the vault (patterns like `0xffff...` allowances and ERC777 'operators' are being used more and more), this presents the risk that someone from the DAO may trigger the function to deposit more tokens than intended by the owner. 

If we leave it like this, then the owner of the tokens must always send the message for the vault to pull the tokens.

Reply:
This would only be for ETH, so no allowances are necessary.

Reply:
I agree this check is not needed for ETH, but we should have the `require(msg.sender == from)` check for token transfers to avoid the scenario I explain in my previous comment

Reply:
I agree this check is not needed for ETH, but we should have the require(msg.sender == from) check for token transfers to avoid the scenario I explain in my previous comment

Reply:
With https://github.com/aragon/aragon-apps/pull/292#discussion_r196114741, I'm hesitating on this too. I think we can leave this as-is so that the vault logs who was the last "hop" in sending it money.